homeassistant/components/mqtt/vacuum/schema_legacy.py

```diff
@@ -363,26 +387,27 @@ def battery_icon(self):
             battery_level=self.battery_level, charging=self._charging
         )
 
-    async def async_turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the vacuum on."""
         if self.supported_features & VacuumEntityFeature.TURN_ON == 0:
             return
-
+        assert self._command_topic is not None
         await self.async_publish(
             self._command_topic,
             self._payloads[CONF_PAYLOAD_TURN_ON],
-            self._qos,
-            self._retain,
-            self._encoding,
+            qos=self._qos,
+            retain=self._retain,
+            encoding=self._encoding,
         )
         self._attr_status = "Cleaning"
         self.async_write_ha_state()
 
-    async def async_turn_off(self, **kwargs):
+    async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the vacuum off."""
         if self.supported_features & VacuumEntityFeature.TURN_OFF == 0:
             return None
 
+        assert self._command_topic is not None
```

Comment:
You could avoid the assert by simply exiting early (applies to all subsequent commands):
```suggestion
        if (
            self._command_topic is None
            or self.supported_features & VacuumEntityFeature.TURN_OFF == 0
        ):
            return None

```